### PR TITLE
Use externalid if available

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

File: jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmSecretStoreBase.java [hunk 1]
```diff
@@ -86,7 +93,6 @@ public void deconstruct() {
 
     // Only for testing
     public Set<AwsRolePath> clientRoleNames() {
```
Comment (gjoranv) [inline jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmSecretStoreBase.java:95]: Suggest to return a set of AssumedRoleInfo (and rename, I see I forgot) and add verification in AsmTenantSecretReaderTest:74

File: jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java [hunk 2]
```diff
@@ -32,31 +34,35 @@ public AsmTenantSecretReader(AsmSecretConfig secretConfig,
...
-        if ( ! vaultIds.containsKey(vault)) {
+        if ( ! vaults.containsKey(vault)) {
             throw new IllegalArgumentException("No vault id found for " + vault);
         }
-        return AthenzUtil.awsReaderRole(system, tenant, vaultIds.get(vault));
+        return AthenzUtil.awsReaderRole(system, tenant, vaults.get(vault).vaultId());
+    }
+
+    protected ExternalId externalId(VaultName vaultName) {
+        return Optional.ofNullable(vaults.get(vaultName)).map(Vault::externalId).orElse(null);
```
Comment (gjoranv) [inline jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java:65]: Shouldn't this throw if externalId is empty?

File: jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java [hunk 3]
```diff
@@ -75,4 +81,5 @@ public static String getAwsSecretId(String system, String tenant, Key key) {
                                           key.vaultName().value(), key.secretName().value());
     }
 
+    record Vault(VaultId vaultId, VaultName vaultName, ExternalId externalId) {}
```
Comment (gjoranv) [inline jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java:84]: Suggest to rename to VaultInfo, and import the subclass in the test.

**Review by gjoranv — APPROVED** [review]

File: jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java (hunk 2 above)
Comment (tokle) [reply jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmTenantSecretReader.java:65]: Yes, let's do in next iteration

**Pull request merged**